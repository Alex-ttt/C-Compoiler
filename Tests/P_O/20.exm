if

    a
  >
    b

then

  Statements:
    for (;;) ..
    First action:
    none
    Stop condition:
    none
    Step:
    none
    Body:
      Statements:
        while () ..
        Condition:
            i
          >
            j
        Body:
          Statements:
            i
            ++
            --
              j
        a
        --
        b
        ++
      Symbols Table:
                  a int
                  b float
                  int function fo(
                          a int
                          b float
                   )
            Statements:
                a
              =
                b
                b
              =
                0
