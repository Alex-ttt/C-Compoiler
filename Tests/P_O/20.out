 Symbols Table:
   a int
   b int
   i int
   j int
   main  int function main(
  )
    {
      Statements:
      if

          a
        >
          b

      then

        {
          Statements:
          for (;;) ..
          First action:
          Stop condition:
          Step:
          Body:
            {
              Statements:
              while () ..
              Condition:
                  i
                >
                  j
              Body:
                {
                  Statements:
                  i
                  ++
                  --
                    j
                }
              Symbols Table:
                              a int
                              b float
            }
          a
          --
          b
          ++
        }
    }
